source/backend/api/Repositories/Mayan/MayanDocumentRepository.cs

```diff
@@ -71,18 +78,26 @@ public async Task<ExternalResponse<DocumentTypeModel>> TryUpdateDocumentTypeAsyn
             _logger.LogDebug("Updating document type...");
 
             string authenticationToken = await _authRepository.GetTokenAsync();
-            JsonSerializerOptions serializerOptions = new()
-            {
-                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
-            };
-            string serializedDocumentType = JsonSerializer.Serialize(documentType, serializerOptions);
+            string serializedDocumentType = JsonSerializer.Serialize(documentType, SerializerOptions);
             using HttpContent content = new StringContent(serializedDocumentType);
             content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
 
             Uri endpoint = new($"{_config.BaseUri}/document_types/{documentType.Id}/");
 
             var response = await PutAsync<DocumentTypeModel>(endpoint, content, authenticationToken);
 
+            if(response.Status == ExternalResponseStatus.Error)
+            {
+                _logger.LogDebug("Error while updating a document type");
+
+                if (_config.ExposeErrors.HasValue && _config.ExposeErrors.Value)
```

Comment:
how would you feel about internalizing this behaviour into the mayan exception itself?